geowebcache/wms/src/main/java/org/geowebcache/service/wms/WMSTileFuser.java

```diff
@@ -272,19 +272,26 @@ protected WMSTileFuser(TileLayerDispatcher tld, StorageBroker sb, HttpServletReq
         
         ImageMime firstMt = null;
         
-        if(iter.hasNext()){
-            firstMt = (ImageMime)iter.next();
-        }
-        boolean outputGif = outputFormat.getInternalName().equalsIgnoreCase("gif");
+        boolean outputJpeg = outputFormat.getInternalName().equalsIgnoreCase("jpeg");
         while (iter.hasNext()) {
-            MimeType mt = iter.next();           
-            if(outputGif){
-                if (mt.getInternalName().equalsIgnoreCase("gif")) {
+            MimeType mt = iter.next();
+
+            // Sets first here so that it will be part of loop
+            if (firstMt == null) {
+                firstMt = (ImageMime) mt;
+            }
+
+            if (outputJpeg) {
+                // use png for stitching to jpeg
+                if (mt.getFileExtension().equalsIgnoreCase("png")) {
```

Comment:
Using an arbitrary PNG subtype?  It seems like preference should be given to truecolour PNG over indexed png Internal names `png` vs `png8`  Using the extension will admittedly ensure you get one of the PNG types, but it is possible, if an unusal decision on the part of the server admin, that both are available.